js/imageeditor/shared/layers/Layers.svelte

```diff
@@ -0,0 +1,193 @@
+<script lang="ts">
+	import { createEventDispatcher, getContext, onMount, tick } from "svelte";
+	import { DropdownArrow } from "@gradio/icons";
+	import { click_outside } from "../utils/events";
+	import { layer_manager, type LayerScene } from "./utils";
+	import { EDITOR_KEY, type EditorContext } from "../ImageEditor.svelte";
+	import type { FileData } from "@gradio/client";
+
+	let show_layers = false;
+
+	export let layer_files: (FileData | null)[] | null = [];
+
+	const dispatch = createEventDispatcher<{
+		new_layer: void;
+	}>();
+
+	const { pixi, current_layer, dimensions } =
+		getContext<EditorContext>(EDITOR_KEY);
+
+	const LayerManager = layer_manager();
+	let layers: LayerScene[] = [];
```

Comment:
We may need to expose more than just the current layers because it causes issues with undo of the current layer isn't the layer that the original command was performed on. Either than or figure out some way for commands to keep track of the current layer at the time.